### PR TITLE
✨ Add option to reset database

### PR DESCRIPTION
**Pull request by sinclairpaul**
# Proposed Changes

As the addon is designed to make the Nginx setup simple, it can be confusing that the database is not removed on uninstall.

Adds config option to force the drop of the database if exists.

Also will assist in the cases of lost passwords etc, I did consider adding a backup option prior, but not really sure it would be needed.

### CODE REVIEW

**Comment by frenck** [comment]
Nice!

I believe an add-on can modify its own options via the Supervisor API. Maybe we can turn it off again automatically?

**Comment by sinclairpaul** [comment]
Added reset of options, in this case its easiest to just clear them, as the risk is we just clear a custom log level.  Maybe a new feature for bashio in the future?

File: proxy-manager/rootfs/etc/cont-init.d/mysql.sh [hunk 1]
```diff
@@ -22,6 +22,19 @@ password=$(bashio::services "mysql" "password")
...
+if bashio::config.true 'reset_database'; then
+    bashio::log.warning 'Recreating database'
+    echo "DROP DATABASE IF EXISTS nginxproxymanager;" \
+    | mysql -h "${host}" -P "${port}" -u "${username}" -p"${password}"
+    
+    #Reset addon options to null
+    curl -s -H 'X-Supervisor-Token: '"$SUPERVISOR_TOKEN"'' \
+        -H 'Content-Type: application/json'  \
+            http://supervisor/addons/self/options \
+	    --data-binary '{"options": {}}'
```
Comment (frenck) [inline proxy-manager/rootfs/etc/cont-init.d/mysql.sh:35]:
```suggestion
    bashio::api.supervisor POST "/addons/self/options" '{"options": {}}'
```

Comment (sinclairpaul) [reply proxy-manager/rootfs/etc/cont-init.d/mysql.sh:35]: Guess I missed that method...

Comment (frenck) [reply proxy-manager/rootfs/etc/cont-init.d/mysql.sh:35]: Yes, probably better to merge the PR for bashio you made, however, that is another cycle. This reduceses it at least a bit for now.

Comment (sinclairpaul) [reply proxy-manager/rootfs/etc/cont-init.d/mysql.sh]: The bashio PR needs some more work, as from what I remember everything will be a string in the state it is in at the moment.  Updated to existing method.

**Review by frenck — APPROVED** [review]
Thanks, @sinclairpaul 👍

**Pull request merged**